Sources/AudioKit/MIDI/MIDI+VirtualPorts.swift

```diff
@@ -160,30 +118,34 @@ extension MIDI {
         destroyVirtualOutputPort()
     }
 
-    /// Closes the virtual input port, if created one already.
+    /// Closes the virtuals input ports, if created one already.
     ///
-    /// - Returns: Returns true if virtual input closed.
+    /// - Returns: Returns true if virtuals inputs closed.
     ///
     @discardableResult public func destroyVirtualInputPort() -> Bool {
-        if virtualInputs[0] != 0 {
-            if MIDIEndpointDispose(virtualInputs[0]) == noErr {
-                virtualInputs[0] = 0
-                return true
+        if virtualInputs != [0] {
+            for (index, virtualInput) in virtualInputs.enumerated().reversed() {
+                guard MIDIEndpointDispose(virtualInput) == noErr else {return false}
+                virtualInputs.remove(at: index)
+            }
+            virtualInputs.append(0)
+            return true
             }
-        }
         return false
     }
 
-    /// Closes the virtual output port, if created one already.
+    /// Closes the virtuals output ports, if created one already.
```

Comment:
I can fix this later, but virtual isn't pluralizable.

Reply:
Done. Sorry for that, english isn't my mother tongue